File: TransCoder_tokenized_test_set_functions/cpp/WAYS_SUM_N_USING_ARRAY_ELEMENTS_REPETITION_ALLOWED.cpp
int countWays ( int arr [ ], int m, int N ) {
  int count [ N + 1 ];
  memset ( count, 0, sizeof ( count ) );
  count [ 0 ] = 1;
  for ( int i = 1;
  i <= N;
  i ++ ) for ( int j = 0;
  j < m;
  j ++ ) if ( i >= arr [ j ] ) count [ i ] += count [ i - arr [ j ] ];
  return count [ N ];
}

File: TransCoder_tokenized_test_set_functions/python/WAYS_SUM_N_USING_ARRAY_ELEMENTS_REPETITION_ALLOWED.py
def countWays ( arr, m, N ) :
    count = [ 0 for i in range ( N + 1 ) ]
    count [ 0 ] = 1
    for i in range ( 1, N + 1 ) :
        for j in range ( m ) :
            if ( i >= arr [ j ] ) :
                count [ i ] += count [ i - arr [ j ] ]
    return count [ N ]
